core/trino-main/src/test/java/io/trino/operator/scalar/TestDateTimeFunctions.java

```diff
@@ -197,6 +201,10 @@ public void testFromUnixTimeNanos()
 
         assertThat(assertions.function("from_unixtime_nanos", "DECIMAL '-12345678900123456789.500'"))
                 .matches("TIMESTAMP '1578-10-13 17:18:03.876543210 Pacific/Apia'");
+
+        assertTrinoExceptionThrownBy(assertions.function("from_unixtime_nanos", "DECIMAL '123456789123456789000000000'")::evaluate)
```

Comment:
> > hey @jkylling,
> > Have added the fixes for `from_iso8601_timestamp` and `from_unixtime_nanos`.
> > In the case of `from_unixtime_nanos` for long, the exception thrown is a `ParsingException` because of BIGINT overflow.
> > ```
> > trino> SELECT from_unixtime_nanos(9223372036854775808);
> > Query 20231215_144131_00019_3i5sr failed: line 1:1: Invalid numeric literal: 9223372036854775808
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > What should be done in this case ? I have ignored it for now since queries like `SELECT 9223372036854775808` also produce the same error.
> 
> Could you add a test with
> 
> ```
> SELECT from_unixtime_nanos(cast('123456789012345678901234567890' as decimal(38,0)));
> ```
> 
> ? Then we won't trigger an exception in the parser, but an exception from the function.


@jkylling I have already added a similar test here. Hope this suffices.

Reply:
That's perfect!